### DISCUSSION TITLE
Segment image with different slice length

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Project-MONAI/MONAI#2218] [Q&A]
Greetings. Is there any way to segment or classify data having images of different slice length (z-axis).
I have the following images' shape 
(224,224,110),(224,224,113),(224,224,100),(224,224,107).
Is there any way, without croping or padding it to fix length?

The reason i am asking as we cant crop/pad the test set, as it not available in most competition.

**Comment by Nic-Ma** [comment]
Hi @talhaanwarch ,

Thanks for your interest here.
I think you can run the inference with batch_size=1 to solve your problem.

Thanks.

**Comment by talhaanwarch** [comment]
But I want to do training. Is it still hold true for training, I mean can I
train volumetric image of different slice length

**Reply by Nic-Ma** [reply]
If you want to batch your images in PyTorch, the images must have same size. So in order to support images with different size, you can use batch_size=1, it would slower but can support your case.
I would suggest you to pad / crop / resize to unify the image size during training.

Thanks.

**Reply by talhaanwarch** [reply]
Thanks @Nic-Ma , unifying is the best option but it will fix the input size and I can't unifying the test data, because of it wouldnot match the mask then

**Reply by rijobro** [reply]
For the test data, you can either crop the mask with the same parameters as the input image, or you can continue using batch_size=1, and then you won't have to crop.
